### ISSUE TITLE
toUrlPiece for NominalDiffTime loses precision

### ISSUE THREAD

**Issue by seagreen** [fizruk/http-api-data#80]
Additionally, `fromUrlPiece` for `NominalDiffTime` can only accept integers, so it can't represent all `NominalDiffTime`s.

Since `NominalDiffTime` uses `Pico` from `Data.Fixed` [underneath](http://hackage.haskell.org/package/time-1.9.2/docs/src/Data.Time.Clock.Internal.NominalDiffTime.html#NominalDiffTime), perhaps this can be changed once https://github.com/fizruk/http-api-data/issues/77 is implemented?

**Comment by rockbmb** [comment]
PR #87 fixes #77, so if it gets merged this issue can be completed.

**Comment by rockbmb** [comment]
This issue can be easily fixes with [this function](https://hackage.haskell.org/package/time-1.9.1/docs/Data-Time-Clock.html#v:nominalDiffTimeToSeconds) recently introduced to `time`, but it cannot be implemented yet.

It relies on `time-1.9.1`, and trying to use it gives the following problem:

```
In the dependencies for Cabal-2.2.0.1:
    time-1.9.1 from stack configuration does not match >=1.4 && <1.9  (latest
               matching version is 1.8.0.4)
needed due to http-api-data-0.3.10 -> Cabal-2.2.0.1

In the dependencies for directory-1.3.1.5:
    time-1.9.1 from stack configuration does not match >=1.4 && <1.9  (latest
               matching version is 1.8.0.4)
needed due to http-api-data-0.3.10 -> directory-1.3.1.5

In the dependencies for unix-2.7.2.2:
    time-1.9.1 from stack configuration does not match >=1.2 && <1.9  (latest
               matching version is 1.8.0.4)
needed due to http-api-data-0.3.10 -> unix-2.7.2.2
```

The newest version of `directory`, 1.3.1, already uses `time (>=1.4 && <1.10)`, and the unreleased `2.8.0.0` version of `unix` as well. However, the latest published version of `Cabal`, which is `2.4.0.1`, still uses `time (>=1.4.0.1 && <1.9)`.

**Comment by fizruk** [comment]
Indeed. For earlier versions I guess we can use `realToFrac :: NominalDiffTime -> Pico`.

@rockbmb are you willing to submit a PR? 😊 

I think you can use CPP to produce something like this:

```haskell
#if MIN_VERSION_time(1,9,1)
nominalDiffTimeToSeconds
#else
(realToFrac :: NominalDiffTime -> Pico)
#endif
```

@phadej is it okay to use CPP here?

**Comment by rockbmb** [comment]
@fizruk will submit a PR 👍

**Comment by phadej** [comment]
Fixed in #88 #89
